### DISCUSSION TITLE
How to replace classic release variables?

### DISCUSSION THREAD

**Discussion by gowallace** [qetza/replacetokens-task#6] [General]
I have an Azure DevOps pipeline that is using the v3 replace tokens task. I saw that v6 was available, so I updated my task to use it, but there are some Azure Devops variables that are no longer being replaced, e.g. `$(Release.ReleaseName)`, `$(Release.AttemptNumber)`. These are built in variables to Azure DevOps: https://learn.microsoft.com/en-us/azure/devops/pipelines/release/variables?view=azure-devops&tabs=batch

Is there a way to make the v6 task pull these in without having to do them individually? Thanks.

**Comment by qetza** [comment]
Hi @gowallace,
Thank for reporting this issue. I'm using the function `getVariables` from the Microsoft SDK which seems to return only all defaults YAML pipelines variables (https://learn.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml) which doesn't includes release variables 😞
I'm going to do some testing and will add code to force read the missing documented default variables.

In the mean time, as a workaround, you can use the _additionalVariables_ input like this:
```
'Release.ReleaseName': '$(Release.ReleaseName)'
'Release.AttemptNumber': '$(Release.AttemptNumber)'
```

**Comment by qetza** [comment]
Hi @gowallace,
The version 6.0.1 of task which was released fixes your issue: the variable names are now case-insensitive like in the previous versions.

**Reply by gowallace** [reply]
Works great! Thank you so much.

**Reply by qetza** [reply]
Thanks for reporting the issue :smile:
